src/torrent.cpp

```diff
@@ -4971,10 +4971,11 @@ bool is_downloading_state(int const st)
 			position += size;
 
 			// pad files always have priority 0
-			download_priority_t const file_prio
-				= fs.pad_file_at(i) ? dont_download
-				: i >= file_prios.end_index() ? default_priority
-				: file_prios[i];
+			download_priority_t const file_prio = [&](){
```

Comment:
I wonder what happens if I send a PR with a refactor like this :D

Reply:
heh, I'm not super happy about this. I might revert it.